pkg/api/vlabs/validate.go

```diff
@@ -193,31 +157,63 @@ func validateKeyVaultSecrets(secrets []KeyVaultSecrets, requireCertificateStore
 
 // Validate implements APIObject
 func (l *LinuxProfile) Validate() error {
-	if e := validateName(l.AdminUsername, "LinuxProfile.AdminUsername"); e != nil {
-		return e
-	}
-	if len(l.SSH.PublicKeys) != 1 {
-		return errors.New("LinuxProfile.PublicKeys requires only 1 SSH Key")
-	}
-	if e := validateName(l.SSH.PublicKeys[0].KeyData, "LinuxProfile.PublicKeys.KeyData"); e != nil {
-		return e
+	// Don't need to call validate.Struct(l)
+	// It is handled by Properties.Validate()
+	if e := validate.Var(l.SSH.PublicKeys[0].KeyData, "required"); e != nil {
+		return fmt.Errorf("KeyData in LinuxProfile.SSH.PublicKeys cannot be empty string")
 	}
 	if e := validateKeyVaultSecrets(l.Secrets, false); e != nil {
 		return e
 	}
 	return nil
 }
 
+func handleValidationErrors(e validator.ValidationErrors) error {
```

Comment:
There is potential for this. We only need a superset of `handleValidationErrors ` to handle all the scenarios. But the assumption would be vlab and v20170701 will have the same constraint on certain field. However, even they are different we can use version to switch it. So i agree we can consolidate them. Let me do it here.